### DISCUSSION TITLE
Static data members

### DISCUSSION THREAD

**Discussion by jmafc** [hsutter/cppfront#616] [Q&A]
There are a couple of issues, #522 and #322, that discuss the subject topic, but it is unclear to me what is the stance with regards to it.  The latter broached `static` in general but was closed (with a "perhaps we'll reopen it"). The former is open but veers in various directions without clarity.

As I understand, static member functions *are* supported: you just have to declare a function without a `this` parameter. However, I believe many static member functions are written as accessors for static data members, so if the latter can't be declared you are forced to declare/define the entire class with cpp1 syntax, is it not?

**Comment by MaxSagebaum** [comment]
In #522 I am still waiting a little bit on the go/no-go from Herb.

**Reply by JohelEGP** [reply]
The next metafunction should be a Cpp1-`enum` or `enum class`-like
(there's already commented out code for that in `cpp2util.h` and `reflect.h2`),
which should require `static` data members,
as I mentioned before at <https://github.com/hsutter/cppfront/issues/522#issuecomment-1602685992>.
So it should be soon.
